### PR TITLE
Fix completed position

### PR DESCRIPTION
**Pull request by perryibm**

### CODE REVIEW

File: core/invoker/src/main/scala/whisk/core/container/ContainerPool.scala [hunk 1]
```diff
@@ -168,40 +169,51 @@ class ContainerPool(
...
+            val conResult = Try(getContainer(1, myPos, key, () => makeWhiskContainer(action, auth)))
+            completedPosition.next()
+            conResult match {
+                case Success(Cold(con, initResult)) =>
+                    info(this, s"Obtained cold container ${con.containerId.id}")
+                    Some(con, Some(initResult))
+                case Success(Warm(con)) =>
+                    info(this, s"Obtained warm container ${con.containerId.id}")
+                    Some(con, None)
+                case Success(Error(str)) =>
```
Comment (markusthoemmes) [inline core/invoker/src/main/scala/whisk/core/container/ContainerPool.scala]: Why is an error modeled as a `Success`?

Comment (perryibm) [inline core/invoker/src/main/scala/whisk/core/container/ContainerPool.scala]: Expected error (docker misbehaving but we handled it) vs the future unexpectedly throwing an exception we had not thought of.   Different error modes.

File: core/invoker/src/main/scala/whisk/core/container/ContainerPool.scala [hunk 2]
```diff
@@ -215,17 +224,11 @@ class ContainerPool(
         }
         if (positionInLine > available) { // e.g. if there is 1 available, then I wait if I am second in line (positionInLine = 2)
```
Comment (rabbah) [inline core/invoker/src/main/scala/whisk/core/container/ContainerPool.scala]: please fix the comments `// e.g. ...` i think these moved around and now are disconnected and confusing.

File: core/invoker/src/main/scala/whisk/core/container/ContainerPool.scala [hunk 3]
```diff
@@ -168,40 +169,48 @@ class ContainerPool(
...
-                completedPosition.next()
+            val key = ActionContainerId(auth.uuid, action.fullyQualifiedName, action.rev)
+            val myPos = nextPosition.next()
+            info(this, s"""Getting container for ${action.fullyQualifiedName} of kind ${action.exec.kind} with ${auth.uuid}:
+                          | myPos = $myPos
+                          | completed = ${completedPosition.cur}
+                          | slack = ${slack()}
+                          | startingCounter = ${startingCounter.cur}""".stripMargin)
+            val conResult = Try(getContainer(1, myPos, key, () => makeWhiskContainer(action, auth)))
+            completedPosition.next()
```
Comment (rabbah) [inline core/invoker/src/main/scala/whisk/core/container/ContainerPool.scala]: Next commit moves this "above" init? (hint don't squash next commits.)

Comment (perryibm) [inline core/invoker/src/main/scala/whisk/core/container/ContainerPool.scala]: i won't - the plan is 2 commits: semantic preserving refactor followed by init moving

Comment (rabbah) [inline core/invoker/src/main/scala/whisk/core/container/ContainerPool.scala]: thanks.

File: core/invoker/src/main/scala/whisk/core/container/ContainerPool.scala (hunk 1 above)
Comment (perryibm) [inline core/invoker/src/main/scala/whisk/core/container/ContainerPool.scala]: Collapsed now.

File: core/invoker/src/main/scala/whisk/core/container/package.scala [hunk 4]
```diff
@@ -73,12 +73,12 @@ package object container {
     /**
      * The result of trying to obtain a container which is known to exist or to create one.
      * Capacity constraints have been passed by this point so there are no Busy's.
-     * The presence of a RunResult indicates this container was not in cache and had to be initialized.
+     * Initiaizliation is performed later so no field for initReulst here.
```
Comment (rabbah) [inline core/invoker/src/main/scala/whisk/core/container/package.scala]: initReulst typo.

File: core/invoker/src/main/scala/whisk/core/container/ContainerPool.scala [hunk 5]
```diff
@@ -254,24 +258,9 @@ class ContainerPool(
...
      */
-    def getOrMake(key: ActionContainerId, conMaker: () => ContainerResult)(implicit transid: TransactionId): Option[ContainerResult] = {
+    def getOrMake(key: ActionContainerId, conMaker: () => WhiskContainer)(implicit transid: TransactionId): Option[ContainerResult] = {
         retrieve(key) match {
-            case CacheMiss => {
-                conMaker() match { /* We make the container outside synchronization */
-                    // Unfortunately, variables are not allowed in pattern alternatives even when the types line up.
-                    case res @ Cold(con, initResult) =>
-                        this.synchronized {
-                            introduceContainer(key, con).state = State.Active
```
Comment (rabbah) [inline core/invoker/src/main/scala/whisk/core/container/ContainerPool.scala]: does this happen somewhere else now?

File: core/invoker/src/main/scala/whisk/core/container/ContainerPool.scala (hunk 2 above)
Comment (perryibm) [inline core/invoker/src/main/scala/whisk/core/container/ContainerPool.scala]: re-arranged

File: core/invoker/src/main/scala/whisk/core/container/ContainerPool.scala (hunk 5 above)
Comment (perryibm) [inline core/invoker/src/main/scala/whisk/core/container/ContainerPool.scala]: still there now

File: core/invoker/src/main/scala/whisk/core/container/package.scala (hunk 4 above)
Comment (perryibm) [inline core/invoker/src/main/scala/whisk/core/container/package.scala]: fixed

**Comment by perryibm** [comment]
PG2-244 passed.

**Pull request merged**